### PR TITLE
add team logic to gamepad

### PR DESCRIPTION
**Pull request by potch**
The gamepad.js component now reports to the server whenever a tank is spawned, indicating what color a connected gamepad should be draw in.

### CODE REVIEW

File: gamepad-host.js [hunk 1]
```diff
@@ -23,7 +24,7 @@
 
     // TODO: Do not hardcode WS URL.
 
-    var sock = new SockJS('http://localhost:30043/socket');
+    var sock = new SockJS('http://10.252.28.228:30043/socket');
```
Comment (cvan) [inline gamepad-host.js]: revert this change, please

File: gamepad-host.js [hunk 2]
```diff
@@ -47,15 +48,22 @@
 
     sock.onmessage = function(e) {
         console.log('WS message:', e.data);
-        if (e.data.n === 'init') {
+        var data = JSON.parse(e.data);
+        if (data.n === 'init') {
```
Comment (cvan) [inline gamepad-host.js:52]: you could use a `switch` on these, but whatever

File: gamepad-host.js [hunk 3]
```diff
@@ -47,15 +48,22 @@
...
             sock.sendMessage('register.gamepad', player);
         }
+        if (data.n === 'gamepad.color') {
+            color = data.d;
+            moveStick.redraw();
+            aimStick.redraw();
+          }
```
Comment (cvan) [inline gamepad-host.js]: fix indentation of closing brace here

File: gamepad-host.js [hunk 4]
```diff
@@ -145,8 +152,9 @@
         }
 
         function ellipse(cx, cy, rx, ry) {
-            for (var i = 0; i <= 6; i++) {
-                var a = i / 3 * 3.14159;
+            var n = 6;
+            for (var i = 0; i <= n; i++) {
+                var a = i / n * (3.14159 * 2);
```
Comment (cvan) [inline gamepad-host.js:157]: intentional change?

File: gamepad-host.js [hunk 5]
```diff
@@ -218,4 +228,23 @@
         sendData('gamepad', state);
     }
 
+    function launchIntoFullscreen(element) {
+      if(element.requestFullscreen) {
```
Comment (cvan) [inline gamepad-host.js]: space before `(`.

Comment (cvan) [inline gamepad-host.js]: separate commit?

File: gamepad-host.js [hunk 6]
```diff
@@ -218,4 +228,23 @@
...
+        element.requestFullscreen();
+      } else if(element.mozRequestFullScreen) {
+        element.mozRequestFullScreen();
+      } else if(element.webkitRequestFullscreen) {
+        element.webkitRequestFullscreen();
+      } else if(element.msRequestFullscreen) {
+        element.msRequestFullscreen();
+      }
+    }
```
Comment (cvan) [inline gamepad-host.js:241]: it'd be cool to also lock orientation à la

https://github.com/mozilla/galaxy.js-mobile-gamepad/blob/master/src/js/lib/utils.js#L138-L153

File: gamepad.js [hunk 7]
```diff
@@ -14,6 +14,16 @@ pc.script.create('gamepad', function (context) {
         initialize: function () {
             this.client = context.root.getChildren()[0].script.client;
             this.link = context.root.findByName('camera').script.link;
+            this.tank = context.root.findByName('tank');
+            this.teams = context.root.getChildren()[0].script.teams;
+
+
+            this.client.socket.on('tank.new', function (data) {
```
Comment (cvan) [inline gamepad.js:19]: you have `function (` (we have `function(` below; that's consistent with the style elsewhere, if you care

File: gamepad.js [hunk 8]
```diff
@@ -14,6 +14,16 @@ pc.script.create('gamepad', function (context) {
         initialize: function () {
             this.client = context.root.getChildren()[0].script.client;
             this.link = context.root.findByName('camera').script.link;
+            this.tank = context.root.findByName('tank');
+            this.teams = context.root.getChildren()[0].script.teams;
+
+
```
Comment (cvan) [inline gamepad.js]: two newlines. are you insane?

File: gamepad.js [hunk 9]
```diff
@@ -14,6 +14,16 @@ pc.script.create('gamepad', function (context) {
         initialize: function () {
             this.client = context.root.getChildren()[0].script.client;
             this.link = context.root.findByName('camera').script.link;
+            this.tank = context.root.findByName('tank');
```
Comment (cvan) [inline gamepad.js]: you're not using this anymore, so you can remove

File: gamepad-host.js (hunk 4 above)
Comment (potch) [inline gamepad-host.js:157]: yup, easier to tweak.

File: gamepad-host.js (hunk 1 above)
Comment (potch) [inline gamepad-host.js]: ya

File: gamepad-host.js (hunk 3 above)
Comment (potch) [inline gamepad-host.js]: ya

File: gamepad-host.js (hunk 6 above)
Comment (potch) [inline gamepad-host.js:241]: doesn't work in my testing.

File: gamepad.js (hunk 8 above)
Comment (potch) [inline gamepad.js]: well, I've got a long list of ex-lovers, _they'll_ tell you I'm insane.

File: gamepad.js [hunk 10]
```diff
@@ -14,8 +14,16 @@ pc.script.create('gamepad', function (context) {
         initialize: function () {
             this.client = context.root.getChildren()[0].script.client;
             this.link = context.root.findByName('camera').script.link;
+            this.teams = context.root.getChildren()[0].script.teams;
 
-            this.client.socket.on('gamepad', function(data) {
+            this.client.socket.on('tank.new', function (data) {
+              this.client.socket.send('gamepad.color', {
```
Comment (cvan) [inline gamepad.js]: this should be 4 spaces

File: gamepad-host.js [hunk 11]
```diff
@@ -218,4 +228,22 @@
...
+        element.mozRequestFullScreen();
+      } else if (element.webkitRequestFullscreen) {
+        element.webkitRequestFullscreen();
+      } else if (element.msRequestFullscreen) {
+        element.msRequestFullscreen();
+      }
+    }
+
+    function setupScreen() {
+      launchIntoFullscreen(document.body);
```
Comment (cvan) [inline gamepad-host.js]: four spaces

File: gamepad.js (hunk 8 above)
Comment (cvan) [inline gamepad.js]: starbucks lovers!
